tests/unit/s2n_random_test.c

```diff
@@ -737,6 +750,7 @@ struct random_test_case random_test_cases[] = {
      * to use 1 below and in s2n_random_test_case_failure_cb().
      */
     { "Test failure.", s2n_random_test_case_failure_cb, CLONE_TEST_DETERMINE_AT_RUNTIME, 1 },
+    { "Test libcrypto's RAND engine is reset correctly after manual s2n_cleanup()", s2n_random_rand_bytes_after_cleanup_cb, CLONE_TEST_DETERMINE_AT_RUNTIME, EXIT_SUCCESS },
 };
```

Comment:
Do we gain anything by running this test from a forked process, since this isn't related to fork detection?

Reply:
It doesn't have to a *forked* process, but this test does need to be the 1st caller that invokes `s2n_init()`, in order to expose the bug. Here's why:

Normally s2n-tls would clean up its rand engine in an atexit handler. However when atexit is turned off (e.g., [aws-c-io turns it off](https://github.com/awslabs/aws-c-io/blob/main/source/s2n/s2n_tls_channel_handler.c#L189)), s2n-tls would clean up it rand engine when [the "main-thread" calls `s2n_cleanup()`](https://github.com/aws/s2n-tls/blob/main/utils/s2n_init.c#L119). The "main thread", in fact, is just [the 1st caller](https://github.com/aws/s2n-tls/blob/main/utils/s2n_init.c#L59) that calls into `s2n_init()`, in a process. Therefore it is necessary for this test to be the 1st caller. Hence running the test in a forked process seems very convenient for this purpose. I'd be happy to relocate this function if there's a better place for the test.

Reply:
Nope, that makes sense. You need to run your test after cleanup, so you need to be able to call s2n_cleanup() form the main thread and still make assertions afterwards.